File: Ruby/Napakalaki/lib/Monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

#encoding: utf-8

class Monster
  def initialize(name, combatLevel, prize, bc, lc)
    @name = name
    @combatLevel = combatLevel
    @prize = prize
    @bc = bc
    @levelChangeAgainstCultistPlayer = lc
  end
  
  def self.newMonsterNoCultist(aname, acombatLevel, aprize, abc)
    new(aname, acombatLevel, aprize, abc, 0)
  end
  
  def self.newMonsterCultist(aname, acombatLevel, aprize, abc, alc)
    new(aname, acombatLevel, aprize, abc, alc)
  end
  
  attr_reader :name, :combatLevel, :bc, :levelChangeAgainstCultistPlayer
  
  def to_s
    "\nName = #{@name} \nCombatLevel = #{@combatLevel} \nPrize: \n  #{@prize.to_s} \nBadConsequence: \n  #{@bc.to_s}"
  end
  
  def getLevelsGained
    return @prize.level
  end
  
  def getTreasuresGained
    return @prize.treasures
  end
  
  def getCombatLevelAgainstCultistPlayer
    return @combatLevel + @levelChangeAgainstCultistPlayer
  end
  
end

File: Java/Napakalaki/src/napakalaki/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;
/**
 *
 * @author juane619
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize premio;
    private BadConsequence bc;
    private int levelChangeAgainstCultistPlayer;
    
    public Monster(String name, int level, BadConsequence bc, Prize prize){
       this.name= name;
       this.combatLevel= level;
       this.premio= prize;
       this.bc= bc;
       levelChangeAgainstCultistPlayer = 0;
    }
    
    public Monster(String name, int level, BadConsequence bc, Prize prize, int lc){
       this.name= name;
       this.combatLevel= level;
       this.premio= prize;
       this.bc= bc;   
       levelChangeAgainstCultistPlayer = lc;
    }

    public String getName() {
        return name;
    }
    
    public int getLevelsGained(){
        return premio.getLevel();
    }
    
    public int getTreasuresGained(){
        return premio.getTreasures();
    }

    public BadConsequence getBadConsequence() {
        return bc;
    }
    
    public Prize getPrize(){
        return premio;
    }
        
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return levelChangeAgainstCultistPlayer+combatLevel;
    }
    
    @Override
    public String toString(){
        return "\n\nNombre del monstruo= " +name + "\nNivel de combate= "+Integer.toString(combatLevel)+"\nPremio: " +premio.toString()+"\nCASTIGO: " + bc.toString();
    }
}
